### PR TITLE
reduce the sharing dependencies tutorial to a guide

### PR DESCRIPTION
**Pull request by fricklerhandwerk**
the contents do not really warrant a full-blown tutorial

### CODE REVIEW

File: source/guides/recipes/sharing-dependencies.md [hunk 1]
```diff
@@ -1,209 +1,112 @@
...
-The actual build recipe will be in `package.nix` and `default.nix` will import this file to perform the build.
-
-First create a `package.nix` file like this:
+Use the `inputsFrom` attribute to `pkgs.mkShell`:
 
 ```nix
+# default.nix
+let
+  pkgs = import <nixpkgs> {};
+  build = pkgs.callpackage ./build.nix {};
```
Comment (infinisil) [inline source/guides/recipes/sharing-dependencies.md:16]:
```suggestion
  build = pkgs.callPackage ./build.nix {};
```

File: source/guides/recipes/sharing-dependencies.md [hunk 2]
```diff
@@ -1,209 +1,112 @@
...
-  version = "0.0.1";
-  format = "pyproject";
-  src = builtins.path { path = ./.; name = "shared_project_source"; };
-  propagatedBuildInputs = [
-    setuptools-scm
-    flask
-  ];
+  inherit build;
+  shell = pkgs.mkShell {
+    inputsfrom = [ build ];
```
Comment (infinisil) [inline source/guides/recipes/sharing-dependencies.md:21]:
```suggestion
    inputsFrom = [ build ];
```

File: source/guides/recipes/sharing-dependencies.md [hunk 3]
```diff
@@ -1,209 +1,112 @@
...
-Try to build this project by running `nix-build -A build`
+## Complete example
 
-[nix_pills_callpackage]: https://nixos.org/guides/nix-pills/callpackage-design-pattern.html
+Assuming your build is defined in `build.nix`:
 
-## Adding development packages
+```nix
+# build.nix
+{ hello }: hello
```
Comment (infinisil) [inline source/guides/recipes/sharing-dependencies.md:39]: This feels a bit too minimal to get the point across. It should at least have some dependencies listed in e.g. `buildInputs` which one could then observe to now also have in the `nix-shell`

**Review by infinisil — APPROVED** [review]
Generally sounds good, @zmitchell what do you think? You wrote the original text.

File: source/guides/recipes/sharing-dependencies.md (hunk 3 above)
Comment (fricklerhandwerk) [reply source/guides/recipes/sharing-dependencies.md:39]: We could have a nonsensical derivation where the builder returns true, but have hello as a build input... or something like that.

Comment (infinisil) [reply source/guides/recipes/sharing-dependencies.md:39]: How about a `cowsay nix > $out` in a `runCommand`, then you can have cowsay as the dependency :)

**Pull request merged**